### ISSUE TITLE
Island Creation with Schematic

### ISSUE THREAD

**Issue by Victhetiger** [rlf/uSkyBlock#764]
There is always an error upon island creation: http://prntscr.com/bfn18v

Note: I do not need it to find a chest, I placed the items inside the schematic before copying it.

USkyblock Configuration: http://pastebin.com/vSQAtz6k
Schematic: http://ge.tt/3ZNIZNb2
Console: http://pastebin.com/U4z9tJqQ
USkyblock /usb v: http://prntscr.com/bfn1cd

Thank you very much. :)

**Comment by BlackBeltPanda** [comment]
Unless it's been changed since I last looked, I believe the chest is required since it also defines the spawn point of the island.

**Comment by Victhetiger** [comment]
Well, there is a chest on the island.

**Comment by BlackBeltPanda** [comment]
Is it a large schematic? I think the chest has to be something like within 15 blocks of the center of the schematic.

**Comment by Victhetiger** [comment]
I copied the schematic 1 block from the chest, the schematic is 2KB

**Comment by BlackBeltPanda** [comment]
Does it happen if you save the schematic with the chest in the center of the island and use that?

**Comment by Victhetiger** [comment]
I'll test it now.

**Comment by Victhetiger** [comment]
Same error, that didn't work. (Also, thank you for your help)

**Comment by Victhetiger** [comment]
Anyone has an idea on how I could've messed up this schematic? Please

**Comment by dutchy1001** [comment]
I get a BIG red warning page when opening the link to your schematic, is it possible to upload it to [Mega](https://mega.nz) so I can test it that way?

**Comment by Victhetiger** [comment]
Here you go: https://mega.nz/#!kcZxUBgJ Sorry about that.

**Comment by dutchy1001** [comment]
When uploading to Mega you need to see the options "with or without" decryption key. Atm it's with key so I cant get it.

**Comment by Victhetiger** [comment]
Well, it's set to without but still as a key, oh well. This link works: https://mega.nz/#!kcZxUBgJ!CkU86EAuYVifk7oh7gJV7CjYws1dTRbGf_g0DFkql90

**Comment by dutchy1001** [comment]
Schematic work with or without Fawe for me, takes few seconds to spawn with no errors in both cases.

**Comment by Victhetiger** [comment]
Do you have the same configuration as I do? I don't know why it says it can't find the chest.

**Comment by dutchy1001** [comment]
I've used 1.9.4 server and a 1.10 server, both are working, with or without Fawe. 
I did use our latest release v2.6.12. Try using the our last release to see if that fixes it.

**Comment by Victhetiger** [comment]
It did not fix it, I've used both 2.6.12 and 2.6.4. I'm currently using 1.9.4.

**Comment by dutchy1001** [comment]
I did notest you haven't added Multiverse-portals and Multiverse-NetherPortals. This might help when added. Links can be found [here](http://dev.bukkit.org/bukkit-plugins/uskyblock/pages/installation/).
I personally use the dev builds of any depended plugin and I have no problem or errors in console.

**Comment by Victhetiger** [comment]
I've used NetherPortals in the past and as soon as they walk into a portal without a destination it teleports them inside the ground and spams their chat. Which is completely ridiculous since that portal was just a spawn portal.

I'm using something else for portals, FancyPortals.

I was wondering how it would help? It's not required and I don't need neither of those 2 plugins.

**Comment by dutchy1001** [comment]
If you don't try you never know, but it will eliminate the use of them.
Since the addition of the Nether to the plugin portals were needed.

**Comment by Victhetiger** [comment]
Alright, I'll try it and get back to you in a minute.

**Comment by rlf** [comment]
If you check out the FAWE project, you will see there are some issues reg. some schematics. Might be worth checking that out.

**Comment by dutchy1001** [comment]
I'll removed the 2 plugins at mine but it still works as it should

@rlf it works on my test setup with or without Fawe or both mv portals plugins

**Comment by Victhetiger** [comment]
I tried it with both MVNP and MVP and it still says the same error: "Watchdog! Balling out..."

**Comment by dutchy1001** [comment]
set in the config 
```
asyncworldedit:

  # Supports disabling the detection of AWE
  enabled: true
```
yours is set to false

**Comment by Victhetiger** [comment]
I was told to disable that since I'm not using AWE I'm using FAWE. But alright, I can try testing that out to.

**Comment by Victhetiger** [comment]
Still says the same thing, I could try enabling AWE with the latest 2.6.12?

**Comment by dutchy1001** [comment]
No matter what I try or what settings I change the island is spawning as expected.

I think it might be time to setup a test server on your machine and start with a default setup including USB, then if that works add 1 by 1 your other plugins.

**Comment by Victhetiger** [comment]
Alright, I'll try that out.

**Comment by dutchy1001** [comment]
It's a pain to do but if it works with just the depended plugins you know it's a conflict from an other plugin. I not, you might need to re-download your plugins

**Comment by Victhetiger** [comment]
I sure hope that's it :smile:

**Comment by rlf** [comment]
FAWE shares config with AWE, so `asyncworldedit.enabled: true` is required for FAWE support to work.

**Comment by Victhetiger** [comment]
Alright, it is confirmed to be FAWE. I'll try it with the latest version of USkyblock now.

**Comment by Victhetiger** [comment]
Same goes with the latest version. There's also an error: http://prntscr.com/bgv1vb

**Comment by Victhetiger** [comment]
http://hastebin.com/obuhogefap.vbs Here's the text error.

**Comment by dutchy1001** [comment]
there is an update for Fawe [here](http://ci.athion.net/job/FastAsyncWorldEdit/)

**Comment by Victhetiger** [comment]
Isn't that latest FAWE version 1.10 only?

**Comment by dutchy1001** [comment]
no look in the list for FastAsyncWorldEdit-bukkit19-3.5.1-fbb4ae9.jar

**Comment by Victhetiger** [comment]
Uskyblock 2.6.12 + FAWE latest, still the same message.

**Comment by dutchy1001** [comment]
I think I found it.
Your **distance** and **protectionRange** are not set to be "chunk" alligned which is probably the issue (meaning multiplied by 32, 1 chunk on each side). So your setting of 300 should be **320** or **288** then. 

We are looking into this to make it work, in your case with 300, but we need some time for testing this.
For now your fix would be to set it to 320 or 288, up to you to go bigger or smaller. An other tip is to set the spawnsize to "half" the island size, this makes it all fit better around spawn, for you it will be 160 or 144 then. Hope that made sense for you.

**Comment by Victhetiger** [comment]
Alright, although if there's already over a thousand islands created with 300, won't it break something with the old ones?

**Comment by dutchy1001** [comment]
that is tricky then, save side is going smaller. Not sure this give trouble in time when (older)players restart their island. For new islands it will not matter.
You can wait for a fix as we probably will continue to fix this tomorrow if possible, depending on our time.
Last resort is a fresh start, there are some new features added to the later versions.

**Comment by Victhetiger** [comment]
Alright, but if I set the distance and protectionRange to less than the current 300, is it less likely to give me trouble than if I set it to higher than 300?

**Comment by dutchy1001** [comment]
yes, if an old player restarts there isnt enough space to go bigger so it will go 20 blocks into the player next to him/her which will then overlap the isands (a very bad thing). 

After I reproduced the issue it still did give me an island (even when the "watchdog bailing out" message was given), but didn't always teleported me to it. With the command **/is home** I was still able to access the island (client restart might help sometimes). 
Also there is an [v2.6.14](https://github.com/rlf/uSkyBlock/releases/tag/v2.6.14) which has progress broadcast and maybe a small improvement.

**Comment by Victhetiger** [comment]
I'll test it out and I'll see about that distance and protectionRange.

**Comment by dutchy1001** [comment]
do backup your server before messing with the settings, just to be on the save side.

**Comment by Victhetiger** [comment]
Alright, great idea!

**Comment by Victhetiger** [comment]
Ok, it's now fixed. The island creation takes less than a second which is incredible! I'm so amazed right now! Alright, so the only thing that might fudge it all up is that the island is located at let's say "-3444" and the region is called "-3300" which might be a problem. Do you have an ETA on when you could fix that problem, if so I might just set it back to 300 for now.

**Comment by dutchy1001** [comment]
Swaping back and fort is not going to do any good to the server, You could have put the creation and restarting of an island on hold till fixed. 
Sorry I didnt think of that last night to say, but that might be as it was already late for me. If you have a backup then you can always roll back to that point and still put it on hold for the time being.

**Comment by rlf** [comment]
@Victhetiger the region is named from the center coordinates of the island.
The region it self is equally distributed around those coordinates.

If you EVER change the protectionRange and distance after having created islands, you will (quite soon) have a totally defunct world.

Just warning you.

**Comment by Victhetiger** [comment]
Fudging tomatoes, alright I changed it back I hope that won't make my world defunct. But now that I know why island create is so long and there's this Watchdog error, I have to wait for it to be fixed?

**Comment by dutchy1001** [comment]
best option as our main dev has warned us ;)

**Comment by Victhetiger** [comment]
Alright well I hope there's a fix soon since now it doesn't even load the islands anymore, no Watchdog message nothing.

**Comment by Victhetiger** [comment]
Atleast the world won't be defunct!

**Comment by Victhetiger** [comment]
I guess it's already defunct, it doesn't load my island anymore.

**Comment by Victhetiger** [comment]
Backup! Yes!

**Comment by dutchy1001** [comment]
As soon you have the backup running, negate the permissions to create and restart islands, players still can play then but no new/restarts can be done to protect your world.

**Comment by Victhetiger** [comment]
Alright, well I truly hope there's a possibility of fixing my mistake in the configuration.

**Comment by dutchy1001** [comment]
we will have a fix soon, our dev is looking into as we speak.

**Comment by Victhetiger** [comment]
Which will also work for the nether? :grinning: Soon is good, soon is really good. Thank you dev.

**Comment by Victhetiger** [comment]
I don't know if this helps the dev fix it but here's an error I got: 
http://prntscr.com/bh7ht1

**Comment by rlf** [comment]
Is that with FAWE? AWE? or WE?

**Comment by Victhetiger** [comment]
I have FAWE and WE, not sure which one it is. But keep in mind my distance is set to "300".

**Comment by rlf** [comment]
It sort of helped.
This is the problem:
https://github.com/boy0001/FastAsyncWorldedit/blob/master/core/src/main/java/com/sk89q/worldedit/EditSession.java#L530

Meaning, you can raise it with @boy0001 on FAWE
It seems it's missing a NullPointer protection when traversing the history (AFAICS).

**Comment by Victhetiger** [comment]
Ah, I see

**Comment by dutchy1001** [comment]
Just giving you an update, we have been testing about 8 (internal) dev releases without a final result. There has been some improvement but this caused other issue's that needs fixing before we can make a pre-release.

**Comment by Victhetiger** [comment]
Alright, well keep me updated! :)

**Comment by Victhetiger** [comment]
Awyiss, good to know it's considered "A bug". That means there's progress! :smiley:

**Comment by dutchy1001** [comment]
We are getting close to have found a solution for non chunk aligned value's.

**Comment by Victhetiger** [comment]
Hooray! :)

**Comment by rlf** [comment]
We have a working solution for chunk aligned and for distance more than 32 above protectionRange. But unfortunately Fawe still "eats" chunks outside the island for the rest of the cases.

**Comment by Victhetiger** [comment]
Where can I download it, unless you are also looking for fixing the FAWE part of the bug?

**Comment by dutchy1001** [comment]
we only have an internal copy, not ready for public use

**Comment by Victhetiger** [comment]
Alrighty

**Comment by Victhetiger** [comment]
Anything new? It's been a few days now. No rush, just asking by curiousity, you don't have to reply if I'm annoying in any way.

**Comment by dutchy1001** [comment]
Sorry but we haven't found the cause of why it takes part of chunks from the neighbouring islands yet.

**Comment by rlf** [comment]
As far as I can see, we have no clear way to avoid FAWE taking those chunks.
Except avoiding using FAWE in those specific scenarios.

I still miss testing whether AWE shows the same issues (or WE for that matter), it might be something introduced in later versions - but I doubt it.

**Comment by Victhetiger** [comment]
Alright, well could we get @boy0001 to reply and help out with that issue I guess.

**Comment by Victhetiger** [comment]
Well that looks like a fix-ish! :smile:

**Comment by dutchy1001** [comment]
I havent tested this properly yet and still can have some issue's regarding to (partly) chunk deletion of neighbouring islands.

**Comment by rlf** [comment]
It will (AFAIK) fix the issue, but unfortunately, it will break compatibility with nether (which should already be disabled for non-chunk-aligned setups), and skygrid (or alternative world-generators) - which might be an issue for some.

**Comment by Victhetiger** [comment]
Oh no, I was hoping fixing the non-chunk aligned setups would also fix the nether. Shoot :disappointed:  But it's fine I think I can figure out another way to add the nether to my server, just a regular world or something.

**Comment by dutchy1001** [comment]
You can add a normal nether with Multiverse and call the world _**skyworld_nether**_. After the world is setup/generated make sure you set the SCALE to 1.0 (default is 8.0) or portals will not send players back to their own island. 
Use the MV command **when your in the skyworld_nether world** ```mvm set scale 1.0```

**Comment by Victhetiger** [comment]
Alrighty, thank you very much. :smile:

**Comment by Victhetiger** [comment]
Wait a minute, you said it'll fix the issue. Which means, the latest /rlf/uSkyblock/releases/ will work with non-chunk-aligned setups?

**Comment by dutchy1001** [comment]
I'm not sure if v2.6.17 has the fix included ( I seen so many "internal" releases the last few days lol), but were still having some other issue's that need to be resolved before a proper release can be made.

**Comment by rlf** [comment]
It will work "better" with FAWE in non-chunk-aligned setups, yea

**Comment by Victhetiger** [comment]
I got an error for you! It might help you out if you didn't already get that error: http://prntscr.com/bnlzhe

**Comment by Victhetiger** [comment]
Well, I'm going to be resetting my whole server in August since it would be quite crappy not to have the nether. That is all I wanted to say.

**Comment by Victhetiger** [comment]
I've reset the entier server, I no longer need this to work. I'll close this ticket unless still needed.
